### PR TITLE
fix(rbac): add role validation for create role

### LINKED ISSUE #943: [RBAC Plugin] Validation is not working for Create Role API
### Describe the bug

Validation is not working working for the Create Role API and allows for the creation of roles that do not match our format of `role:default/<ROLE_NAME>`. This means that we can create roles like this `xyzzy` or `x:default/my-roles`.

However, we do have validation on the other Role APIs. This means that we end up with errors when calling those APIs

`Unsupported kind x. List supported values [\"user\", \"group\", \"role\"]" name :  "InputError" stack :  "InputError: Unsupported kind x. List supported values [\"user\", \"group\", \"role\"]\n    at PolicesServer.getEntityReference (/Users/dsantra/Downloads/backstage-plugins/plugins/rbac-backend/src/service/policies-rest-api.ts:619:13)\n    at <anonymous> (/Users/dsantra/Downloads/backstage-plugins/plugins/rbac-backend/src/service/policies-rest-api.ts:458:36)\n    at process.processTicksAndRej`

### Expected Behavior

Validation should be the same across all APIs

### What are the steps to reproduce this bug?

1. Set up RBAC backend plugin
2. Assign yourself as an admin
3. Create the role `x:default/my-roles`
4. Attempt to delete the role `x:default/my-roles`

### PR DESCRIPTION
**Pull request by PatAKnight**
## Description

This PR fixes the issue of missing validation on the role name in the POST and PUT methods for roles. This also adds a couple of additional unit tests to get `policies-rest-api.ts` to 100% test coverage.

## Fixes

- Fixes #943 

## How to test changes / Special notes to the reviewer

1. Update app-config to include your user as admin
```yaml
permission:
  enabled: true
  rbac:
    admin:
      users:
        - name: user:default/<USERNAME>
```
2. Log in using GitHub auth
3. Get token and export it in the terminal with `export token=...token...`
4. Use curl commands to test validation

wrong kind
`curl -X POST "http://localhost:7007/api/permission/roles" -d '{ "memberReferences": [ "user:default/test", "user:default/test2" ], "name": "x:default/test" }' -H "Content-Type: application/json" -H "Authorization: Bearer $token" -v`

Missing name
`curl -X POST "http://localhost:7007/api/permission/roles" -d '{ "memberReferences": [ "user:default/test", "user:default/test2" ], "name": "role:default/" }' -H "Content-Type: application/json" -H "Authorization: Bearer $token" -v`

Missing name and namespace
`curl -X POST "http://localhost:7007/api/permission/roles" -d '{ "memberReferences": [ "user:default/test", "user:default/test2" ], "name": "role:" }' -H "Content-Type: application/json" -H "Authorization: Bearer $token" -v`

Missing namespace
`curl -X POST "http://localhost:7007/api/permission/roles" -d '{ "memberReferences": [ "user:default/test", "user:default/test2" ], "name": "role:/test" }' -H "Content-Type: application/json" -H "Authorization: Bearer $token" -v`

Missing kind
`curl -X POST "http://localhost:7007/api/permission/roles" -d '{ "memberReferences": [ "user:default/test", "user:default/test2" ], "name": "default/test" }' -H "Content-Type: application/json" -H "Authorization: Bearer $token" -v`

Missing kind and namespace
`curl -X POST "http://localhost:7007/api/permission/roles" -d '{ "memberReferences": [ "user:default/test", "user:default/test2" ], "name": "test" }' -H "Content-Type: application/json" -H "Authorization: Bearer $token" -v`

Missing kind and name
`curl -X POST "http://localhost:7007/api/permission/roles" -d '{ "memberReferences": [ "user:default/test", "user:default/test2" ], "name": "default/" }' -H "Content-Type: application/json" -H "Authorization: Bearer $token" -v`

### CODE REVIEW

File: plugins/rbac-backend/src/service/policies-rest-api.test.ts [hunk 1]
```diff
@@ -1152,6 +1164,21 @@ describe('REST policies api', () => {
...
+        .post('/roles')
+        .send({
+          memberReferences: ['user:default/permission_admin'],
+          name: 'x:default/rbac_admin',
+        });
+
+      expect(result.statusCode).toBe(400);
+      expect(result.body.error).toEqual({
+        name: 'InputError',
+        message: `Invalid role definition. Cause: Unsupported kind x. List supported values ["user", "group", "role"]`,
```
Comment (AndrienkoAleksandr) [inline plugins/rbac-backend/src/service/policies-rest-api.test.ts:1178]: I think message should be: 
```suggestion
        message: `Invalid role definition. Cause: Unsupported kind x. List supported values ["user", "group", "role"]`,
```
```suggestion
        message: `Invalid role definition. Cause: Unsupported kind x. Supported value should be "role"`,
```

**Comment by AndrienkoAleksandr** [comment]
Sorry, @PatAKnight, I accidentally committed my proposal. Actually I wanted to point you that validator should be fixed to return message about support single 'kind' for role entity.

File: plugins/rbac-backend/src/service/policies-rest-api.test.ts (hunk 1 above)
Comment (AndrienkoAleksandr) [reply plugins/rbac-backend/src/service/policies-rest-api.test.ts]: Ops. I didn't wanted to make commit...

**Comment by PatAKnight** [comment]
> Sorry, @PatAKnight, I accidentally committed my proposal. Actually I wanted to point you that validator should be fixed to return message about support single 'kind' for role entity.

It's not a problem, and I can take a look into updating the validator to return the updated message for roles.

**Comment by PatAKnight** [comment]
Closing in favor of #1020. Was having technical difficulties trying to get this pulled in to my local branch and decided that it would be faster to create a new branch and copy over the code.

**Pull request closed without merging**